src/main/java/teammates/logic/core/SendgridService.java

```diff
@@ -25,30 +29,38 @@
      * {@inheritDoc}
      */
     @Override
-    public Email parseToEmail(EmailWrapper wrapper) {
-        Email email = new Email();
-        email.setFrom(wrapper.getSenderEmail());
-        if (wrapper.getSenderName() != null && !wrapper.getSenderName().isEmpty()) {
-            email.setFromName(wrapper.getSenderName());
+    public Mail parseToEmail(EmailWrapper wrapper) {
+        Mail email = new Mail();
+        Email sender;
+        if (wrapper.getSenderName() == null || wrapper.getSenderName().isEmpty()) {
+            sender = new Email(wrapper.getSenderEmail());
+        } else {
+            sender = new Email(wrapper.getSenderEmail(), wrapper.getSenderName());
         }
-        email.setReplyTo(wrapper.getReplyTo());
-        email.addTo(wrapper.getRecipient());
+        email.setFrom(sender);
+        email.setReplyTo(new Email(wrapper.getReplyTo()));
+        Personalization personalization = new Personalization();
+        personalization.addTo(new Email(wrapper.getRecipient()));
         if (wrapper.getBcc() != null && !wrapper.getBcc().isEmpty()) {
-            email.addBcc(wrapper.getBcc());
+            personalization.addBcc(new Email(wrapper.getBcc()));
         }
+        email.addPersonalization(personalization);
         email.setSubject(wrapper.getSubject());
-        email.setHtml(wrapper.getContent());
-        email.setText(Jsoup.parse(wrapper.getContent()).text());
```

Comment:
Are we no longer setting the plaintext part of the email?

Reply:
I don't think that is necessary in the v3 API